autoload/neomake/makers/ft/elixir.vim

```diff
@@ -1,12 +1,34 @@
 " vim: ts=4 sw=4 et
 
+" Credo error types.
+" F -> Refactoring opportunities
+" W -> Warnings
+" C -> Convention violation
+" D -> Software design suggestions
+" R -> Readability suggestions
+" Map structure {CredoError: NeomakeType, ...}
+let s:neomake_elixir_credo_type_map = {
+    \ 'F': 'W',
+    \ 'C': 'W',
+    \ 'D': 'I',
+    \ 'R': 'I'}
+
 function! neomake#makers#ft#elixir#PostprocessEnforceMaxBufferLine(entry) abort
     let buffer_lines = str2nr(line('$'))
     if (buffer_lines < a:entry.lnum)
         let a:entry.lnum = buffer_lines
     endif
 endfunction
 
+function! neomake#makers#ft#elixir#PostprocessCredo(entry) abort
+    let type = toupper(a:entry.type)
+    let type_map = extend(s:neomake_elixir_credo_type_map,
+                \ get(g:, 'neomake_elixir_credo_type_map', {}))
```

Comment:
@tweekmonster 
> `getcompletion('g:neomake_<ft>_<maker>_config_.*', 'var')`

Where's the dict there?
Wouldn't that rather be `g:neomake_<ft>_<maker>_config` then?

(TIL about `getcompletion()` though - there are some places where this could be useful (e.g. `neomake#Get(Project)Makers`).)

@epilgrim 
It's a topic that came up several times already, sorry to hijack your PR.. ;)
Apart from that I think there should be no unnecessary config options, but if it's there already we might want to keep it for now?!
But it might also be better to have something generic, that allows users to apply such mappings, e.g. a generic postprocess hook like mentioned in https://github.com/neomake/neomake/pull/977.